language/squad/pytorch/train.py

```diff
@@ -0,0 +1,829 @@
+  )
+  parser.add_argument("--learning_rate", default=5e-5, type=float, help="The initial learning rate for Adam.")
+  parser.add_argument(
+    "--gradient_accumulation_steps",
+    type=int,
+    default=1,
+    help="Number of updates steps to accumulate before performing a backward/update pass.",
+  )
+  parser.add_argument("--weight_decay", default=0.0, type=float, help="Weight decay if we apply some.")
+  parser.add_argument("--adam_epsilon", default=1e-8, type=float, help="Epsilon for Adam optimizer.")
+  parser.add_argument("--max_grad_norm", default=1.0, type=float, help="Max gradient norm.")
+  parser.add_argument(
+    "--num_train_epochs", default=3.0, type=float, help="Total number of training epochs to perform."
+  )
+  parser.add_argument(
+    "--max_steps",
+    default=-1,
+    type=int,
+    help="If > 0: set total number of training steps to perform. Override num_train_epochs.",
+  )
+  parser.add_argument("--warmup_steps", default=0, type=int, help="Linear warmup over warmup_steps.")
+  parser.add_argument(
+    "--n_best_size",
+    default=20,
+    type=int,
+    help="The total number of n-best predictions to generate in the nbest_predictions.json output file.",
+  )
+  parser.add_argument(
+    "--max_answer_length",
+    default=30,
+    type=int,
+    help="The maximum length of an answer that can be generated. This is needed because the start "
+         "and end predictions are not conditioned on one another.",
+  )
+  parser.add_argument(
+    "--verbose_logging",
+    action="store_true",
+    help="If true, all of the warnings related to data processing will be printed. "
+         "A number of warnings are expected for a normal SQuAD evaluation.",
+  )
+  parser.add_argument(
+    "--lang_id",
+    default=0,
+    type=int,
+    help="language id of input for language-specific xlm models (see tokenization_xlm.PRETRAINED_INIT_CONFIGURATION)",
+  )
+
+  parser.add_argument("--logging_steps", type=int, default=500, help="Log every X updates steps.")
+  parser.add_argument("--save_steps", type=int, default=500, help="Save checkpoint every X updates steps.")
+  parser.add_argument(
+    "--eval_all_checkpoints",
+    action="store_true",
+    help="Evaluate all checkpoints starting with the same prefix as model_name ending and ending with step number",
+  )
+  parser.add_argument("--no_cuda", action="store_true", help="Whether not to use CUDA when available")
+  parser.add_argument(
+    "--overwrite_output_dir", action="store_true", help="Overwrite the content of the output directory"
+  )
+  parser.add_argument(
+    "--overwrite_cache", action="store_true", help="Overwrite the cached training and evaluation sets"
+  )
+  parser.add_argument("--seed", type=int, default=42, help="random seed for initialization")
+
+  parser.add_argument("--local_rank", type=int, default=-1, help="local_rank for distributed training on gpus")
```

Comment:
I'm not sure how to get distributed working. Have you been able to train on 2 gpus? If so, it'd be nice if there were instructions. The train.py file makes it seem easy to train on multiple gpus.

Reply:
I was not able to make multi-gpu training work. I'll try and make it work in another PR.